lib_device/functions.cpp

```diff
@@ -0,0 +1,21 @@
+#include <magma.h>
+
+// count GPUs function
+
+
+extern "C"  void  MagmaGpuNum_Available(int* max_ngpus)
+{
+    magma_device_t devices[117]; // 117 is large enough to fit all device IDs
```

Comment:
Where is 117 coming from? Also as the value repeats in the getdevices call below, probably should be a variable to keep synchronized if any changes happen.